wizeline/wize-q-remix
# fix: User identity revealed despite posting as anonymous user

#### What does this PR do?
It fixes the bug where the user's identity would be displayed in a question despite the question being posted as anonymous.

https://user-images.githubusercontent.com/104774638/200703982-4fd45fb7-b664-4926-be59-7fe1261af611.mov

#### Where should the reviewer start?
- app/routes/questions/new.jsx

#### How should this be manually tested?
1. Post a new question as an anonymous user.
2. : Verify that the question is indeed anonymous and your account's information is not associated to the newly created question.
3. Post an authored question.
4. Verify your account's information is associated to the newly created question.

#### Any background context you want to provide?

The route for the redirection after creating the question had to be changed to  "/?index" to fix an issue where the toast success notification was being displayed multiple times upon going back and forth the question detail page and the index page after the question creation.

The only logic added is omitting the "created_by_employee_id" property (passing it as null) in the payload to create the question when the "is_anonymous" property is true in the request as it is done in the old backend project.

Reference: https://github.com/wizeline/wize-q-back/blob/b197fde689bdbf82a8e70376ccfe833b7e863be2/controllers/questions/create.js#L22-L38

#### What are the relevant tickets?
(Link to issues, related PR, JIRA issues, etc.)
Closes #73
